### PR TITLE
exif: sanitize datetime on reading metadata

### PR DESCRIPTION
**Pull request by kmilos**
Fixes https://github.com/darktable-org/darktable/issues/9254

### CODE REVIEW

**Comment by johnny-bit** [comment]
Since this re-uses already existing code i think it's safe for 3.6.

File: src/common/image.c [hunk 1]
```diff
@@ -477,7 +477,7 @@ static void _set_datetime(const int32_t imgid, const char *datetime)
   /* fetch image from cache */
   dt_image_t *image = dt_image_cache_get(darktable.image_cache, imgid, 'w');
 
-  memcpy(&image->exif_datetime_taken, datetime, sizeof(image->exif_datetime_taken));
+  g_strlcpy(&image->exif_datetime_taken, datetime, sizeof(image->exif_datetime_taken));
```
Comment (johnny-bit) [inline src/common/image.c:480]: seems like CI doesn't like that one ;)

```
fatal error: incompatible pointer types passing 'char (*)[20]' to parameter of type 'gchar *' (aka 'char *') [-Wincompatible-pointer-types]
```

Comment (kmilos) [reply src/common/image.c:480]: Actually exposed a bug that was there all the time, shouldn't be an addr, it's already a char array...

**Review by TurboGit — APPROVED** [review]
All good to me. Thanks!

**Pull request merged**